app/controllers/CategoryGuidanceController.scala

```diff
@@ -44,21 +44,22 @@ class CategoryGuidanceController @Inject() (
     extends FrontendBaseController
     with I18nSupport {
 
-  def onPageLoad: Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
-    request.userAnswers.get(CommodityQuery) match {
-      case Some(commodity) =>
-        val ottResponseFuture = ottConnector.getCategorisationInfo(commodity.commodityCode)
+  def onPageLoad(recordId: String): Action[AnyContent] = (identify andThen getData andThen requireData).async {
+    implicit request =>
+      request.userAnswers.get(CommodityQuery) match {
+        case Some(commodity) =>
+          val ottResponseFuture = ottConnector.getCategorisationInfo(commodity.commodityCode)
 
-        for {
-          goodsNomenclature  <- ottResponseFuture
-          categorisationInfo <- Future.fromTry(Try(CategorisationInfo.build(goodsNomenclature).get))
-          updatedAnswers     <- Future.fromTry(request.userAnswers.set(CategorisationQuery, categorisationInfo))
-          _                  <- sessionRepository.set(updatedAnswers)
-        } yield Ok(view())
+          for {
+            goodsNomenclature  <- ottResponseFuture
```

Comment:
can you just put this in the for comprehension instead of saving it to a val since it's not re-used